tests/v2/test_0150-flatten.py

```diff
@@ -545,43 +667,119 @@ def test_flatten0():
+    assert ak._v2.operations.structure.flatten(array, axis=0).tolist() == [
         [1, 2, 3],
         [],
         [4, 5],
     ]
-    assert array.flatten(axis=-2).tolist() == [
+    assert ak._v2.operations.structure.flatten(array, axis=-2).tolist() == [
         [1, 2, 3],
         [],
         [4, 5],
     ]
 
     array = ak._v2.highlevel.Array([1, 2, 3, 4, 5]).layout
 
-    assert array.flatten(axis=0).tolist() == [1, 2, 3, 4, 5]
-    assert array.flatten(axis=-1).tolist() == [1, 2, 3, 4, 5]
+    assert ak._v2.operations.structure.flatten(array, axis=0).tolist() == [
+        1,
+        2,
+        3,
+        4,
+        5,
+    ]
+    assert ak._v2.operations.structure.flatten(array, axis=-1).tolist() == [
+        1,
+        2,
+        3,
+        4,
+        5,
+    ]
+
 
+def test_flatten_allow_regulararray_size_zero_0590():
 
-@pytest.mark.skip(reason="ak.flatten() not implemented yet")
-def test_flatten_allow_regulararray_size_zero():
     empty = ak._v2.contents.RegularArray(
         ak._v2.highlevel.Array([[1, 2, 3], [], [4, 5]]).layout, 0, zeros_length=0
     )
 
-    assert empty.flatten(axis=0).tolist() == []
-    assert empty.flatten(axis=1).tolist() == []
-    assert empty.flatten(axis=2).tolist() == []
+    assert ak._v2.operations.structure.flatten(empty, axis=0).tolist() == []
+    assert ak._v2.operations.structure.flatten(empty, axis=1).tolist() == []
+    assert ak._v2.operations.structure.flatten(empty, axis=2).tolist() == []
+
+
+@pytest.mark.skip(reason="ak.repartition() not implemented yet")
+def test_0612():
+    assert ak._v2.operations.structure.flatten(
+        ak.repartition(ak.Array([[0, 1, 2], [], [3, 4], [5], [6, 7, 8, 9]]), 3)
+    ).tolist() == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+
+@pytest.mark.skip(reason="ak.repartition() not implemented yet")
+def test_0724():
+    a = ak._v2.contents.NumpyArray(np.empty(0))
+    idx = ak._v2.index.Index64([])
+    a = ak._v2.contents.IndexedOptionArray(idx, a)
+    idx = ak._v2.index.Index64([0])
+    a = ak._v2.contents.ListOffsetArray(idx, a)
+    idx = ak._v2.index.Index64([175990832])
+    a = ak._v2.contents.ListOffsetArray(idx, a)
+
+    assert ak._v2.operations.structure.flatten(a, axis=2).tolist() == []
+    assert (
+        str(ak._v2.operations.structure.flatten(a, axis=2).type) == "0 * var * ?float64"
+    )
+
+
+@pytest.mark.skip(reason="ak.where() not implemented yet")
+def test_flatten_axis_none_0866():
+    a1 = ak.zip({"a": [[1], [], [2, 3]], "b": [[4], [], [5, 6]]}, with_name="a1")
+    a2 = ak.zip({"a": [[7, 8], [9], []], "b": [[10, 11], [12], []]}, with_name="a2")
+    union = ak.where([True, False, True], a1, a2)
+    assert set(ak.flatten(union, axis=None)) == set([1, 2, 3, 4, 5, 6, 9, 12])
+
+
+@pytest.mark.skip(reason="ak.from_numpy not implemented yet")
+def test_0973():
+    array = ak.from_numpy(np.zeros((3, 3, 5)))
+    flattened = ak.flatten(array, axis=None)
+    assert flattened.ndim == 1
```

Comment:
`ak.from_numpy` is implemented. Try it:

```suggestion
def test_0973():
    array = ak._v2.operations.convert.from_numpy(np.zeros((3, 3, 5)))
    flattened = ak._v2.operations.structure.flatten(array, axis=None)
    assert flattened.ndim == 1
```